apple/internal/xcarchive.bzl

```diff
@@ -33,7 +33,8 @@ def _xcarchive_impl(ctx):
     arguments.add("--bundle", bundle_info.archive.path)
     arguments.add("--output", xcarchive.path)
 
-    for dsym in dsym_info.direct_dsyms:
+    dsyms = dsym_info.transitive_dsyms.to_list()
```

Comment:
Instead of popping the `depset`, we should use `arguments.add_all()` with `before_each = "--dsym"`.

Reply:
How about `expand_directories = False`?

Reply:
TIL, thanks